src/core/cheats.c

```diff
@@ -802,3 +803,30 @@ void mCheatDeviceDeinit(struct mCPUComponent* component) {
 		}
 	}
 }
+
+void mCheatToggle(struct mCheatDevice* device) {
+	size_t i;
+	bool cheatsDisabled = true;
+	for (i = 0; i < mCheatSetsSize(&device->cheats); ++i) {
+		struct mCheatSet* cheat = *mCheatSetsGetPointer(&device->cheats, i);
+		if (cheat->enabled) {
+			_unpatchROM(device, cheat);
+			cheat->enabled = false;
+			cheat->savedToggle = true;
+			cheatsDisabled = false;
+		} else if (device->resetSaved) {
+			cheat->savedToggle = false;
+		}
+	}
+	device->resetSaved = false;
+	if (cheatsDisabled) {
```

Comment:
If all the cheats are disabled, then we want to re-enable any cheats that we saved.